src/System.Runtime.Extensions/src/System/Environment.Unix.cs

```diff
@@ -112,21 +114,209 @@ private static IDictionary GetEnvironmentVariablesCore(EnvironmentVariableTarget
+                    string data = GetEnvironmentVariable("XDG_DATA_HOME");
+                    if (string.IsNullOrEmpty(data))
                     {
-                        return home;
+                        data = Path.Combine(home, ".local", "share");
                     }
+                    return data;
+
+                case SpecialFolder.Desktop:
+                case SpecialFolder.DesktopDirectory:
+                    return ReadXdgDirectory(home, "XDG_DESKTOP_DIR", "Desktop");
+                case SpecialFolder.Templates:
+                    return ReadXdgDirectory(home, "XDG_TEMPLATES_DIR", "Templates");
+                case SpecialFolder.MyVideos:
+                    return ReadXdgDirectory(home, "XDG_VIDEOS_DIR", "Videos");
+
+                case SpecialFolder.MyMusic:
+                    return IsMac ? Path.Combine(home, "Music") : ReadXdgDirectory(home, "XDG_MUSIC_DIR", "Music");
+                case SpecialFolder.MyPictures:
+                    return IsMac ? Path.Combine(home, "Pictures") : ReadXdgDirectory(home, "XDG_PICTURES_DIR", "Pictures");
+                case SpecialFolder.Fonts:
+                    return IsMac ? Path.Combine(home, "Library", "Fonts") : Path.Combine(home, ".fonts");
+
+                case SpecialFolder.Favorites:
+                    if (IsMac) return Path.Combine(home, "Library", "Favorites");
+                    break;
+                case SpecialFolder.InternetCache:
+                    if (IsMac) return Path.Combine(home, "Library", "Caches");
                     break;
+            }
+
+            // No known path for the SpecialFolder
+            return string.Empty;
+        }
+
+        private static string GetXdgConfig(string home)
+        {
+            // "$XDG_CONFIG_HOME defines the base directory relative to which user specific configuration files should be stored."
+            // "If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used."
+            string config = GetEnvironmentVariable("XDG_CONFIG_HOME");
+            if (string.IsNullOrEmpty(config))
+            {
+                config = Path.Combine(home, ".config");
+            }
+            return config;
+        }
+
+        private static string ReadXdgDirectory(string homeDir, string key, string fallback)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(homeDir), $"Expected non-empty homeDir");
+            Debug.Assert(!string.IsNullOrEmpty(key), $"Expected non-empty key");
+            Debug.Assert(!string.IsNullOrEmpty(fallback), $"Expected non-empty fallback");
 
-                // TODO: Add more special folder handling
+            string envPath = GetEnvironmentVariable(key);
+            if (!string.IsNullOrEmpty(envPath))
+            {
+                return envPath;
+            }
+
+            string userDirsPath = Path.Combine(GetXdgConfig(homeDir), "user-dirs.dirs");
```

Comment:
> Let's keep it simple and follow what Mono does

Sounds reasonable to me (since it's now a conscious choice).  It's probably worth putting a comment in that says it's using a simplistic model that doesn't use application overrides, fallbacks, et cetera.